site/src/pages/WorkspaceBuildPage/WorkspaceBuildPageView.tsx

```diff
@@ -23,38 +37,238 @@ const sortLogsByCreatedAt = (logs: ProvisionerJobLog[]) => {
 }) => {
+  const styles = useStyles()
+
+  if (!build) {
+    return <Loader />
+  }
+
   return (
-    <Margins>
-      {build && (
-        <PageHeader>
-          <Stack direction="row" alignItems="center" spacing={3}>
-            <BuildAvatar build={build} size="xl" />
-            <div>
-              <PageHeaderTitle>Build #{build.build_number}</PageHeaderTitle>
-              <PageHeaderSubtitle condensed>
-                {build.initiator_name}
-              </PageHeaderSubtitle>
-            </div>
-          </Stack>
-        </PageHeader>
-      )}
-
-      <Stack spacing={4}>
-        {build &&
-          build.transition === "delete" &&
-          build.job.status === "failed" && (
-            <WorkspaceBuildStateError build={build} />
+    <Box
+      sx={{
+        height: "calc(100vh - 62px - 36px)",
+        overflow: "hidden",
+        // Remove padding added from dashboard layout (.siteContent)
+        marginBottom: "-48px",
+        display: "flex",
+        flexDirection: "column",
+      }}
+    >
+      <FullWidthPageHeader sticky={false}>
+        <Stack direction="row" alignItems="center" spacing={3}>
+          <BuildAvatar build={build} />
+          <div>
+            <PageHeaderTitle>Build #{build.build_number}</PageHeaderTitle>
+            <PageHeaderSubtitle>{build.initiator_name}</PageHeaderSubtitle>
+          </div>
+        </Stack>
+
+        <Stats aria-label="Build details" className={styles.stats}>
+          <StatsItem
+            className={styles.statsItem}
+            label="Workspace"
+            value={
+              <Link
+                to={`/@${build.workspace_owner_name}/${build.workspace_name}`}
+              >
+                {build.workspace_name}
+              </Link>
+            }
+          />
+          <StatsItem
+            className={styles.statsItem}
+            label="Template version"
+            value={build.template_version_name}
+          />
+          <StatsItem
+            className={styles.statsItem}
+            label="Duration"
+            value={displayWorkspaceBuildDuration(build)}
+          />
+          <StatsItem
+            className={styles.statsItem}
+            label="Started at"
+            value={new Date(build.created_at).toLocaleString()}
+          />
+          <StatsItem
+            className={styles.statsItem}
+            label="Action"
+            value={
+              <Box component="span" sx={{ textTransform: "capitalize" }}>
+                {build.transition}
+              </Box>
+            }
+          />
+        </Stats>
+      </FullWidthPageHeader>
+
+      <Box
+        sx={{
+          display: "flex",
+          alignItems: "start",
+          overflow: "hidden",
+          flex: 1,
+        }}
+      >
+        <Sidebar>
+          <SidebarCaption>Builds</SidebarCaption>
+          {!builds &&
+            [...Array(15).keys()].map((i) => (
+              <BuildSidebarItemSkeleton key={i} />
+            ))}
```

Comment:
`Array.from({ length: 15 }, (_, i) => ...)` does the same thing but allocates less and is a bit nicer imo :)

Reply:
TIL